File: nutrican-be/src/main/java/com/sba/nutricanbe/chat/service/ChatService.java
package com.sba.nutricanbe.chat.service;

import com.sba.nutricanbe.chat.dto.ChatMessageRequest;
import com.sba.nutricanbe.chat.dto.ChatMessageResponse;
import com.sba.nutricanbe.chat.dto.ChatThreadResponse;
import com.sba.nutricanbe.chat.dto.UpdateChatMessageRequest;
import com.sba.nutricanbe.common.dto.ApiResponse;
import com.sba.nutricanbe.common.dto.PageResponse;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.UUID;

public interface ChatService {
    ApiResponse<List<ChatThreadResponse>> getThreads(UUID userId);

    ApiResponse<PageResponse<ChatMessageResponse>> getMessages(UUID userId, UUID mappingId, int page, int size);

    ChatMessageResponse sendMessage(UUID senderId, ChatMessageRequest request);

    ChatMessageResponse sendImageMessage(UUID senderId, UUID mappingId, String content, MultipartFile file);

    ChatMessageResponse sendAttachmentMessage(UUID senderId, UUID mappingId, String content, MultipartFile file,
            String contextType, UUID contextRefId);

    ChatMessageResponse updateMessage(
            UUID userId, UUID mappingId, UUID messageId, UpdateChatMessageRequest request);

    void deleteMessage(UUID userId, UUID mappingId, UUID messageId);

    ApiResponse<Void> markRead(UUID userId, UUID mappingId);
}

File: nutrican-fe/src/services/chatService.js
import api from './api';

export const chatService = {

    getThreads: () => {
        return api.get('/chat/threads');
    },


    getMessages: (mappingId, params = { page: 0, size: 30 }) => {
        return api.get(`/chat/threads/${mappingId}/messages`, { params });
    },

    sendImage: (mappingId, file, content) => {
        const formData = new FormData();
        formData.append('file', file);
        if (content?.trim()) {
            formData.append('content', content.trim());
        }
        return api.post(`/chat/threads/${mappingId}/images`, formData);
    },


    markRead: (mappingId) => {
        return api.put(`/chat/threads/${mappingId}/read`);
    },

    sendAttachment: (mappingId, file, content, contextType, contextRefId) => {
        const formData = new FormData();
        formData.append('file', file);
        if (content?.trim()) formData.append('content', content.trim());
        if (contextType) formData.append('contextType', contextType);
        if (contextRefId) formData.append('contextRefId', contextRefId);
        return api.post(`/chat/threads/${mappingId}/attachments`, formData);
    },

    updateMessage: (mappingId, messageId, content) =>
        api.patch(`/chat/threads/${mappingId}/messages/${messageId}`, { content }),

    deleteMessage: (mappingId, messageId) =>
        api.delete(`/chat/threads/${mappingId}/messages/${messageId}`),
};
